app/controllers/admin/organizations_controller.rb

```diff
@@ -30,12 +30,28 @@ def index
 
   def new
     @organization = Organization.new
-    @organization.users.build(organization_admin: true)
+    account_request = params[:token] && AccountRequest.get_by_identity_token(params[:token])
+
+    if account_request.blank?
+      @organization.users.build(organization_admin: true)
+    elsif account_request.processed?
+      flash[:error] = "The account request had already been processed and cannot be used again"
+      @organization.users.build(organization_admin: true)
+    else
+      @organization.assign_attributes({
+                                        name: account_request.organization_name,
+                                        url: account_request.organization_website,
+                                        email: account_request.email,
+                                        account_request_id: account_request.id
+                                      })
+      @organization.users.build(organization_admin: true, email: account_request.email, name: account_request.name)
```

Comment:
We approached a similar problem with `Request` -> `Distribution` before and I believe there was a method on `Distribution` called something like `:from_request` or something along those lines. You could probably extract this logic to a method on `Organization` or allow an `AccountRequest` object to be passed in or something. 

Not a blocker, but just an idea to clarify the flow here since controller flows can be easily cross-polluted

Reply:
@armahillo good idea! I'll integrate a similar method here.